actionsystemtest/lib/action_system_test/driver_adapters/web_server.rb

```diff
@@ -0,0 +1,45 @@
+begin
+  require "rack/handler/puma"
+rescue LoadError
+  false
+end
+
+module ActionSystemTest
+  module DriverAdapters
+    module WebServer # :nodoc:
```

Comment:
Let's turn this from a concern into a real class. It relies on shared state where it's getting mixed in and it has its own private methods. Too much entanglement to be a good candidate for a concern, imo.